client/client.go

```diff
@@ -485,6 +505,7 @@ func newHttpsClient(conf Config) (*http.Client, error) {
 type HttpsClient struct {
 	Certificate string
 	Key         string
+	SSLEngine   string
```

Comment:
SSLEngine is uninitialized when calling `openssl.EngineById`, will openSSL handle this as "the default"? Or what is the logic here?

Reply:
if the SSLEngine is empty it will just fail. The logic here is: that if you configure `Key` to be a PKCS#11 URI you have to provide some engine. I can check that in `Validate` and output error if engine is not provided. is that ok?

Reply:
Yes, I think we should check it for a nicer error to the user.